src/ui/ImageRenderer/index.tsx

```diff
@@ -99,77 +87,118 @@ const ImageRenderer = ({
+          borderRadius: getBorderRadiusForImageRenderer(circle, borderRadius),
         }}
       />
     );
-  }, [internalUrl]);
-
-  return dynamicMinWidth && dynamicMinHeight && (
-    <div
-      className={[
-        ...(Array.isArray(className) ? className : [className]),
-        'sendbird-image-renderer',
-      ].join(' ')}
-      style={{
-        width: '100%',
-        minWidth: dynamicMinWidth,
-        maxWidth: fixedSize ? dynamicMinWidth : '400px',
-        height: dynamicMinHeight,
-      }}
-      ref={ref}
-    >
-      {showPlaceHolder && PlaceHolder}
-      {
-        showDefaultComponent
-          ? DefaultComponent
-          : (
-            <div
-              className="sendbird-image-renderer__image"
-              style={{
-                width: '100%',
-                minWidth: dynamicMinWidth,
-                maxWidth: fixedSize ? dynamicMinWidth : '400px',
-                height: dynamicMinHeight,
-                position: 'absolute',
-                backgroundRepeat: 'no-repeat',
-                backgroundPosition: 'center',
-                backgroundSize: 'cover',
-                backgroundImage: `url(${internalUrl})`,
-                borderRadius: getBorderRadiusForImageRenderer(circle, borderRadius),
-              }}
-            />
-          )
-      }
-      {HiddenImageLoader}
-      {
-        shadeOnHover && <div
-          className="sendbird-multiple-files-image-renderer__image-cover"
-          style={{
-            ...{ borderRadius: getBorderRadiusForImageRenderer(circle, borderRadius) },
-            ...(isUploaded ? {} : { display: 'inline-flex' }),
+  };
+
+  return (
+    dynamicMinWidth
+    && dynamicMinHeight && (
+      <div
+        ref={ref}
+        className={[...(Array.isArray(className) ? className : [className]), 'sendbird-image-renderer'].join(' ')}
+        style={{
+          width: '100%',
+          minWidth: dynamicMinWidth,
+          maxWidth: fixedSize ? dynamicMinWidth : '400px',
+          height: dynamicMinHeight,
+        }}
+      >
+        {placeholderVisible && renderPlaceholder()}
+        {defaultComponentVisible ? renderDefault() : renderImage()}
+        {shadeOnHover && (
+          <div
+            className="sendbird-multiple-files-image-renderer__image-cover"
+            style={{
+              ...{ borderRadius: getBorderRadiusForImageRenderer(circle, borderRadius) },
+              ...(isUploaded ? {} : { display: 'inline-flex' }),
+            }}
+          />
+        )}
+        <HiddenImageLoader
+          src={internalUrl}
+          alt={alt}
+          onLoadStart={() => {
+            setPlaceholderVisible(true);
+            setDefaultComponentVisible(false);
+          }}
+          onLoad={() => {
+            setPlaceholderVisible(false);
+            setDefaultComponentVisible(false);
+            onLoad();
+          }}
+          onError={() => {
+            setDefaultComponentVisible(true);
```

Comment:
nit: doesn't need to keep displaying the placehodler so...
```suggestion
            setPlaceholderVisible(true);
            setDefaultComponentVisible(true);
```

Reply:
nice catch thank you